### DISCUSSION TITLE
Proposal: Add an index / TOC with years to the blog posts archive page

### DISCUSSION THREAD

**Discussion by ilg-ul** [facebook/docusaurus#9950] [Ideas]
I was thinking perhaps we can improve UX when accessing the Archive page, by adding an index / TOC with years?

Here is an example of a site with many years, where reaching the old entries requires a lot of scrolling:

- <https://cronica-it.github.io/evenimente/arhiva/>

and another page in the same site (there are 3 blog instances):

- <https://cronica-it.github.io/amintiri/arhiva/>

My first thought was a regular TOC, on the right side of the windows, but I'm fine with any other idea (on the top?).

**Comment by slorber** [comment]
The standard Docusaurus TOC highlights TOC items when you scroll down. I'm not sure this kind of UX makes sense with a 3-columns layout, and I feel like this is a feature you need but not sure others need it to.

You can swizzle the component and build the UX you want yourself. If there is enough interest in our user base I'm ok to consider adding but, but I'd prefer to see userland implementations first and see how they differ.

- If at least 3 people swizzle that page, add a custom TOC, and deploy it to production, I will consider it's worth it to study the common denominator between the 3+ solutions, and add this to Docusaurus. 
- If only you deploys it to production, it means there's likely not enough traction to implement this feature in Docusaurus.

**Reply by slorber** [reply]
@johnnyreilly you also have a large blog with many years, would you also find this useful?

https://johnnyreilly.com/blog

**Reply by johnnyreilly** [reply]
It's funny, I'm somewhat on the fence. It might be nice yes. But I also think it's fine without it. No strong feelings! 😀

**Reply by ilg-ul** [reply]
> The standard Docusaurus TOC highlights TOC items when you scroll down. I'm not sure this kind of UX makes sense with a 3-columns layout

That's why I asked for comments, since I'm not sure too. :-(

Would a simple paragraph full of years, on the top, be better?

> You can swizzle the component and build the UX you want yourself.

I can swizzle the component, but my html+css knowledge is far from enough for designing a proper solution, and definitely not enough for implementing it :-(
